modules/kargoAnalyticsAdapter.js

```diff
@@ -1,14 +1,96 @@
+import { logError } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
 import adapter from '../src/AnalyticsAdapter.js';
 import adapterManager from '../src/adapterManager.js';
+import CONSTANTS from '../src/constants.json';
 
-var kargoAdapter = adapter({
-  analyticsType: 'endpoint',
-  url: 'https://krk.kargo.com/api/v1/event/prebid'
-});
+const EVENT_URL = 'https://krk.kargo.com/api/v1/event';
+const KARGO_BIDDER_CODE = 'kargo';
+
+const analyticsType = 'endpoint';
+
+let _initOptions = {};
+
+let _logBidResponseData = {
+  auctionId: '',
+  auctionTimeout: 0,
+  responseTime: 0,
+};
+
+let _bidResponseDataLogged = [];
+
+var kargoAnalyticsAdapter = Object.assign(
+  adapter({ analyticsType }), {
+    track({ eventType, args }) {
+      switch (eventType) {
+        case CONSTANTS.EVENTS.AUCTION_INIT: {
+          _logBidResponseData.auctionTimeout = args.timeout;
+          break;
+        }
+        case CONSTANTS.EVENTS.BID_RESPONSE: {
+          handleBidResponseData(args);
+          break;
+        }
+      }
+    }
+  }
+);
+
+// handleBidResponseData: Get auction data for Kargo bids and send to server
+function handleBidResponseData (bidResponse) {
+  // Verify this is Kargo and we haven't seen this auction data yet
+  if (bidResponse.bidder !== KARGO_BIDDER_CODE || _bidResponseDataLogged.includes(bidResponse.auctionId) !== false) {
+    return
+  }
+
+  _logBidResponseData.auctionId = bidResponse.auctionId;
+  _logBidResponseData.responseTime = bidResponse.timeToRespond;
```

Comment:
so having this object at module scope we could possibly see issues if two auctions are running at same time.

The right thing would prob be to just create a local object based off your wanted params here.

Even the `auctionTimeout` part may change between auctions.

Now it is probably fine majority of the time. So I can be fine with leaving it if you want.

Let me know.